src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java

```diff
@@ -465,10 +466,10 @@ else if (!TokenUtil.areOnSameLine(startToken, brace)) {
      * Validate EOL case.
      *
      * @param brace brace AST
-     * @param braceLine line content
+     * @param braceLine content of line with brace in array of unicode code points
      */
-    private void validateEol(DetailAST brace, String braceLine) {
-        if (CommonUtil.hasWhitespaceBefore(brace.getColumnNo(), braceLine)) {
+    private void validateEol(DetailAST brace, int...braceLine) {
```

Comment:
Usage of varargs in all related updates (https://github.com/checkstyle/checkstyle/pull/11095, https://github.com/checkstyle/checkstyle/pull/11096, and others) has been on my mind.

Consider:

```bash

➜  cat Test.java
public class Test {
    static void m(int... varargs) {
        System.out.println(varargs);
    }

    public static void main(String[] args) {
        m();
        m(1, 2, 3);
    }
}
➜  javac Test.java
➜ 

```
API of method using varargs allows zero or more arguments to be passed, and we can pass list of `int`s, etc.

This is not good, and could lead to problems in the future. Incorrect usage/ assumptions about method with varargs will only surface in failure at runtime, or as a fault.

```bash

➜  cat Test.java  
public class Test {
    static void m(int[] arr) {
        System.out.println(arr);
    }

    public static void main(String[] args) {
        m();
        m(1, 2, 3);
    }
}
➜  javac Test.java
Test.java:7: error: method m in class Test cannot be applied to given types;
        m();
        ^
  required: int[]
  found:    no arguments
  reason: actual and formal argument lists differ in length
Test.java:8: error: method m in class Test cannot be applied to given types;
        m(1, 2, 3);
        ^
  required: int[]
  found:    int,int,int
  reason: actual and formal argument lists differ in length
2 errors

```

API of method using `int[]` demands primitive array of `int`s.

Incorrect usage of method with `int[]` parameter will be caught at compile time or caught in IDE. This saves engineer time, and resolves any doubts about API of given method.

I think we should remove usage of varargs in all related updates.

@pbludov @strkkk what do you think?

Reply:
I think exactly the opposite - if it is possible to use varargs, they should be used instead of array.

varargs is an array, it is just a syntax sugar to make code less verbose. Consider
```
method1(int[] args) {}
method2(int... args) {}
```
invocations will be like
```
method1(new int[0]);
method1(new int[] {1, 2, 3})
```
versus
```
method2();
method2(1, 2, 3);
```

second approach is way clearer for me. Let compiler do its job and create array itself

Reply:
For reference, I believe there is a CI item (Maybe TeamCity?) that encourages `...` over `[]` for parameters if it is the final one.

Since the change you want is to make the code more strict that the array is required and not optional, it makes sense to me if the circumstance is right. If some parameter was TRULY required, such a method should throw an exception either through NPE or IAE in some fashion either through direct code or implied.

However, it matters more if the Utility's parameter is required or just a recommendation like `...` recommends. I think each instance of usage matters and would need to be examined "Why does the code work if an empty array is given?".

Reply:
@strkkk I think your logic is sound, thanks for explanation.

> `method2(1, 2, 3)` 

This may be an improvement over `method1(new int[] {1, 2, 3})` (less verbose, more obvious); I think it is the ability to do `method2()` and implicitly initialize an array that bothers me mostly.

I am interested to see what others think.

Reply:
PMD fails if array is last paramenter and varargs is not used.
If method uses varargs and we pass no argument, then when we try to access its element, method will throw OutOfBound Exception.
So, there will not be a case of miss or bug cond.

Reply:
> PMD fails if array is last paramenter and varargs is not used.

https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#usevarargs

It seems like developers like the flexibility of varargs; this is indisputable.

Resolved.